generator/booking_dataset.cpp

```diff
@@ -306,58 +308,34 @@ void BookingDataset::LoadHotels(istream & src, string const & addressReferencePa
   }
 }
 
-bool BookingDataset::MatchWithBooking(OsmElement const & e) const
+size_t BookingDataset::MatchWithBooking(FeatureBuilder1 const & fb) const
 {
-  string name;
-  for (auto const & tag : e.Tags())
-  {
-    if (tag.key == "name")
-    {
-      name = tag.value;
-      break;
-    }
-  }
+  auto const name = fb.GetName(StringUtf8Multilang::kDefaultCode);
 
   if (name.empty())
     return false;
 
+  auto const center = MercatorBounds::ToLatLon(fb.GetKeyPoint());
   // Find |kMaxSelectedElements| nearest values to a point.
   auto const bookingIndexes =
-      GetNearestHotels(e.lat, e.lon, kMaxSelectedElements, kDistanceLimitInMeters);
-
-  bool matched = false;
+      GetNearestHotels(center.lat, center.lon, kMaxSelectedElements, kDistanceLimitInMeters);
 
   for (size_t const j : bookingIndexes)
   {
-    if (booking_scoring::Match(GetHotel(j), e).IsMatched())
-      break;
+    if (booking_scoring::Match(GetHotel(j), fb).IsMatched())
+      return j;
   }
 
-  return matched;
+  return numeric_limits<size_t>::max();
 }
 
-bool BookingDataset::Filter(OsmElement const & e,
-                            function<bool(OsmElement const &)> const & fn) const
+bool BookingDataset::CanBeBooking(FeatureBuilder1 const & fb) const
```

Comment:
Sorry, I didn't get the idea why this method is called CanBeBooking. The method is just checks that fb has a hotel description with a non-empty default name. But the condition is necessary, not sufficient, right?